### PR TITLE
replace path querying of id w/ fetching user data in context

### PR DESCRIPTION
**Pull request by jerrydngzh**
- that + refactoring where appropriate
- alot of line changes was prettier autoformat -> i have max line width at 100 instead of default 80

### CODE REVIEW

File: frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
- // @ts-nocheck
-import React, { useState, useEffect } from 'react';
-import { MapContainer, TileLayer, Marker, Popup, useMapEvents } from 'react-leaflet';
-import L from 'leaflet';
-import markerIcon from '/images/marker-icon.png';
+//@ts-nocheck
```
Comment (Toastjpg) [inline frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx:1]: leaving //@ts-nocheck?

File: frontend/src/components/CreateMemo/page.tsx [hunk 2]
```diff
@@ -109,20 +65,19 @@ const CreateMemo = ({}) => {
...
-      // console.log(res)
-      const newMemoId = res._id; // TODO
+      const createdMemo = await MemoController.create_memo(currentUser.uid, memoToCreate);
+      const newMemoId = createdMemo._id; // TODO
 
       // Update the memos array in userData with the new memo ID
       let newUserData: any = userData;
       newUserData.memos = [...userData.memos, newMemoId];
 
       setUserData(newUserData);
```
Comment (Toastjpg) [inline frontend/src/components/CreateMemo/page.tsx:75]: Shouldnt need to set user data upon creation of memo. Page will navigate to dashboard upon success. On failure of `UserController.update_user `, the UserData state on the create memo page should not be affected.

File: frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
 import { UserController } from "../../../../controllers/user.controller";
 import { MemoController } from "../../../../controllers/memo.controller";
-const SavedLocations = ({
-  id,
-  reloadDropdown,
-  onDropdownReloaded,
-  onLocationSelected,
-}) => {
+
+// FIXME -- the entire component function, remove `//@ts-nocheck` and fix issues
```
Comment (Toastjpg) [inline frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx:6]: lol

**Review by Toastjpg — APPROVED** [review]
LGTM. Approving refactoring of id from query param to Auth Provider context. Looks very clean and removes alot of unnecessary code. 

Noticed alot of FIXMEs tho LOL

File: frontend/src/components/CreateMemo/page.tsx (hunk 2 above)
Comment (jerrydngzh) [reply frontend/src/components/CreateMemo/page.tsx:75]: just realized this has revealed a bug and has to do with a problem with both the logic on the frontend and design of our backend services.

see #57 but also #58

**Pull request merged**